app/Services/Invoice/AutoBillInvoice.php

```diff
@@ -82,7 +83,7 @@ public function run()
         }
 
         /* $gateway fee */
-        $fee = $gateway_token->gateway->calcGatewayFee($amount, $this->invoice->uses_inclusive_taxes);
+        $fee = $gateway_token->gateway->calcGatewayFee($amount, GatewayType::CREDIT_CARD, $this->invoice->uses_inclusive_taxes);
```

Comment:
for discussion, i don't think this is right, from the token we should be able to determine its type and inject that? for instance in the gateway is a BANK_TRANSFER then we would need to apply that fee instead of CREDIT_CARD?

Reply:
Sorry, but can you guide me on this one? Say, for example, Stripe is a gateway token. How would you pull the gateway type?